### PR TITLE
Fix and test uploading file with name separated by spaces

### PR DESCRIPTION
**Pull request by sergey-chechaev**
Fix error when uploading file with name separated by spaces

Example error:

`ExAws.S3.put_object("bucket-test", "some name with spaces.txt", "data") |> ExAws.request!`

Response: 

```
** (ExAws.Error) ExAws Request Error!

{:error, {:http_error, 403, %{body: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message><Key></Key><BucketName></BucketName><Resource>/bucket-test/some+name+with+spaces.txt</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>", headers: [{"Accept-Ranges", "bytes"}, {"Content-Type", "application/xml"}, {"Server", "Minio/RELEASE.2017-01-25T03-14-52Z (linux; amd64)"}, {"Vary", "Origin"}, {"X-Amz-Request-Id", "14A31A0FEFC6C04E"}, {"Date", "Tue, 14 Feb 2017 08:29:17 GMT"}, {"Transfer-Encoding", "chunked"}], status_code: 403}}}

(ex_aws) lib/ex_aws.ex:46: ExAws.request!/2
```

### CODE REVIEW

**Comment by sergey-chechaev** [comment]
Hey @benwilson512, can we merge it?

File: test/lib/ex_aws/s3/upload_test.exs [hunk 1]
```diff
@@ -22,6 +22,14 @@ defmodule ExAws.S3.UploadTest do
       assert_received :chunk_uploaded
       assert_received :completed_upload
     end
+
+    test "uploading a file with a name separated by spaces" do
+      %{status_code: status_code} =
+        ExAws.S3.put_object("sergey-test", "some file name.txt", "data")
+        |> ExAws.request!
```
Comment (benwilson512) [inline test/lib/ex_aws/s3/upload_test.exs]: I can't accept tests like this at this time. I don't have one such bucket, and I'm very very hesitant about giving the AWS keys associated with ExAws's travis account the ability to write objects to buckets.

Comment (sergey-chechaev) [inline test/lib/ex_aws/s3/upload_test.exs]: I changed `sergey-test` bucket to `my-bucket` like in test: `test "uploading a file with a stream"`

**Comment by KernelMadness** [comment]
@benwilson512 PR seems legit, what do you think about merging it?

**Comment by lessless** [comment]
@sergey-chechaev I would recommend dropping `mix.lock` from the PR

**Comment by benwilson512** [comment]
I have included the `request.ex` change in master, I didn't want to merge to mix.lock changes and tests involving uploading data to S3. The credentials that run the test suite do not have permission to upload data.

**Pull request closed without merging**